modules/ppcp-button/services.php

```diff
@@ -65,25 +67,86 @@
 
 		return $dummy_ids[ $shop_country ] ?? $container->get( 'button.client_id' );
 	},
-	'button.smart-button'                         => static function ( ContainerInterface $container ): SmartButtonInterface {
+	'button.is_paypal_continuation'               => static function( ContainerInterface $container ): bool {
+		$session_handler = $container->get( 'session.handler' );
+
+		$order = $session_handler->order();
+		if ( ! $order ) {
+			return false;
+		}
+		$source = $order->payment_source();
+		if ( $source && $source->card() ) {
+			return false; // Ignore for DCC.
+		}
+		if ( 'card' === $session_handler->funding_source() ) {
+			return false; // Ignore for card buttons.
+		}
+
+		return true;
+	},
+	'button.context'                              => static function( ContainerInterface $container ): string {
```

Comment:
This does not match `ContextTrait`.

Can you remove this service and use `ContextTrait` instead? Because putting this into a service may not be a good idea anyway, some of these checks may fail if called too early.